parquet/src/bin/parquet-concat.rs

```diff
@@ -66,47 +66,55 @@ impl Args {
 
         let output = File::create(&self.output)?;
 
-        let inputs = self
-            .input
-            .iter()
-            .map(|x| {
-                let reader = File::open(x)?;
-                let metadata = ParquetMetaDataReader::new().parse_and_finish(&reader)?;
-                Ok((reader, metadata))
-            })
-            .collect::<Result<Vec<_>>>()?;
+        let schema = {
```

Comment:
Since there are no tests, I think a comment here explaining the rationale for not keeping the files open is probably good

```suggestion
        // Check schemas in a first pass to make sure they all match
        // and then do the work in a second pass after the validatin
        let schema = {
```